pkg/cache/memory/repository.go

```diff
@@ -266,6 +267,54 @@ func (r *cachedRepository) update(ctx context.Context, updated repository.Packag
 	// TODO: Just updated package?
 	identifyLatestRevisions(r.cachedPackageRevisions)
 
+	if updated.Lifecycle() == (v1alpha1.PackageRevisionLifecyclePublished) {
+		updatedMain := updated.ToMainPackageRevision()
```

Comment:
A comment before line 270 explaining that this if block is fixing up the main branch would be helpful. Or better still if you could refactor lines 271-316 into a function, out might help the understandability of the code.

Reply:
I've moved that now into a function :D